### PR TITLE
Add cmesh example cubed spherical shell  [3/n].

### PR DESCRIPTION
**Pull request by jmark**
**_Describe your changes here:_**

This is a follow-PR to https://github.com/DLR-AMR/t8code/pull/787.

grafik

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

#### General
- [ ] The reviewer executed the new code features at least once and checked the results manually

- [ ] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [ ] New source/header files are properly added to the Makefiles
- [ ] The code is well documented
- [ ] All function declarations, structs/classes and their members have a proper doxygen documentation
- [ ] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)

#### Tests
- [ ] The code is covered in an existing or new test case using Google Test

#### Github action

- [ ] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)
- [ ] All tests pass (in various configurations, this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

#### Scripts and Wiki

- [ ] If a new directory with source-files is added, it must be covered by the `script/find_all_source_files.scp` to check the indentation of these files.
- [ ] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

#### Licence

- [ ] The author added a BSD statement to `doc/` (or already has one)

### CODE REVIEW

**Comment by jmark** [comment]
@sandro-elsweijer I assigned you this PR since it is nearly identical to https://github.com/DLR-AMR/t8code/pull/787 and should not add much workload to review it.

**Review by sandro-elsweijer — CHANGES_REQUESTED** [review]
I would find it better to adapt the `t8_cmesh_new_quadrangulated_spherical_surface` routine from your previous PR. We could add a 2D/3D flag and then just add another loop for the radial layers.
The approach to make an independent forest seems to be overkill and would involve more code maintenance in the future

File: src/t8_cmesh/t8_cmesh_examples.c [hunk 1]
```diff
@@ -2910,3 +2913,102 @@ t8_cmesh_new_quadrangulated_spherical_surface (const double radius, sc_MPI_Comm
...
+                                    const int num_layers, sc_MPI_Comm comm)
+{
+  /* Initialization of the mesh */
+  t8_cmesh_t cmesh;
+  t8_cmesh_init (&cmesh);
+
+  t8_geometry_c *geometry = t8_geometry_cubed_spherical_shell_new ();
+  t8_cmesh_register_geometry (cmesh, geometry);
+
+  const int two_to_nl = num_levels > 0 ? 1 << num_levels : 1; /* 2^num_levels */
```
[the remark this answers is not in the document]
Comment (jmark) [reply src/t8_cmesh/t8_cmesh_examples.c:2928]: Actually, you are rigtht! Fixed!

File: example/cmesh/t8_cmesh_geometry_examples.cxx [hunk 2]
```diff
@@ -163,7 +163,28 @@ main (int argc, char **argv)
...
+    const char *prefix_cmesh = "t8_cubed_spherical_shell_cmesh";
+    const char *prefix_forest = "t8_cubed_spherical_shell_forest";
+
+    const int uniform_level = 1;
+    const double inner_radius = T8_SQRT3;
+    const double shell_thickness = 0.2;
+    const int num_levels = 3;
+    const int num_layers = 2;
+
+    t8_cmesh_t cmesh = t8_cmesh_new_cubed_spherical_shell (inner_radius, shell_thickness, num_levels, num_layers, comm);
```
[the remark this answers is not in the document]
Comment (jmark) [reply example/cmesh/t8_cmesh_geometry_examples.cxx:176]: Do we really want to rename this? "Cubed sphere" is actually an established term.

File: src/t8.h [hunk 3]
```diff
@@ -116,6 +116,9 @@ typedef uint64_t t8_linearidx_t;
 /** Define precisions for computations */
 #define T8_PRECISION_EPS SC_EPS
 
+/** Cube squared. */
```
[the remark this answers is not in the document]
Comment (jmark) [reply src/t8.h:119]: Maybe. Removed it.

**Review by sandro-elsweijer — APPROVED** [review]

**Pull request merged**